### PR TITLE
Update pyproject.toml to bump cryptography  version minimum

### PR DESCRIPTION
**Pull request by pixitha**
Bumping cryptography to 42 minimum, when they added the `not_valid_before_utc` function to the cryptography library.

not_valid_before_utc[](https://cryptography.io/en/latest/x509/reference/#cryptography.x509.Certificate.not_valid_before_utc)

    Added in version 42.0.0.

Was getting an error when it was using version 41.0.7 on my machine.

### CODE REVIEW

**Comment by fulder** [comment]
Thank you for your contribution @pixitha, you're right it should be bumped to the minimum version. Just don't forget to update the `poetry.lock` file as well.

File: poetry.lock [hunk 1]
```diff
@@ -2,38 +2,39 @@
 
 [[package]]
 name = "annotated-types"
-version = "0.5.0"
+version = "0.7.0"
```
Comment (fulder) [inline poetry.lock:5]: Looks like a lot of dependencies were updated, did you maybe run `poetry lock`  without the `--np-update` flag? See: https://python-poetry.org/docs/cli/#options-11

Comment (pixitha) [reply poetry.lock]: I misunderstood your initial suggestion to update the poetry file, as I needed to run the update command, oops. I've never worked with poetry before to be honest. reverted and changed to the  `poetry lock --no-update`  command

**Review by fulder — APPROVED** [review]

File: poetry.lock (hunk 1 above)
Comment (fulder) [reply poetry.lock]: No problem, I will merge this and create a release shortly :)

**Pull request merged**